plugins/core/IO/qLASIO/src/LasScalarFieldLoader.cpp

```diff
@@ -0,0 +1,403 @@
+//##########################################################################
+//#                                                                        #
+//#                CLOUDCOMPARE PLUGIN: LAS-IO Plugin                      #
+//#                                                                        #
+//#  This program is free software; you can redistribute it and/or modify  #
+//#  it under the terms of the GNU General Public License as published by  #
+//#  the Free Software Foundation; version 2 of the License.               #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#                   COPYRIGHT: Thomas Montaigu                           #
+//#                                                                        #
+//##########################################################################
+
+#include "LasScalarFieldLoader.h"
+
+#include <laszip/laszip_api.h>
+
+#include <ccPointCloud.h>
+#include <ccScalarField.h>
+
+#include <utility>
+
+// TODO take by move
+LasScalarFieldLoader::LasScalarFieldLoader(std::vector<LasScalarField> standardScalarFields,
+                                           std::vector<LasExtraScalarField> extraScalarFields,
+                                           ccPointCloud &pointCloud)
+    : m_standardFields(std::move(standardScalarFields)), m_extraScalarFields(std::move(extraScalarFields))
```

Comment:
Aren't you moving the already copied vectors this way? Shouldn't you take the vectors as references (or even double &&?). Or is it on purpose that you want to copy these vectors? (in which case I don't see the point of using std::move).

And if the aim is to have only one instance of these vectors, I don't really see the benefit compared to managing references of the input vectors in this class?

Reply:
Yes this is moving the copy of the vectors from the arguments to the member variable.
Without the std::move, we would be copying the vector from the argiments to the member variable, introducing an unneeded extra copy step.


But I agree that in case we can simply use simple references / directly move with &&

Reply:
Yes, even passing the vector as const references would only end in a single copy.